src/offset.js

```diff
@@ -82,27 +82,28 @@ jQuery.fn.extend({
 				});
 		}
 
-		var docElem, win,
+		var docElem, win, rect,
 			elem = this[ 0 ],
-			box = { top: 0, left: 0 },
 			doc = elem && elem.ownerDocument;
 
 		if ( !doc ) {
 			return;
 		}
 
-		docElem = doc.documentElement;
+		rect = elem.getBoundingClientRect();
 
-		// Make sure it's not a disconnected DOM node
-		if ( !jQuery.contains( docElem, elem ) ) {
-			return box;
+		// Make sure element is not hidden (display: none) or disconnected
+		if ( !rect.width && !rect.height && !rect.left &&
+			!rect.top && !rect.right && !rect.bottom ) {
```

Comment:
This is overkill; the only properties that matter for detecting visibility are width and height—and note that there is literally no way to differentiate no-content inline elements that happen to be at viewport origin from disconnected or display:none elements with gBCR, since they all have zero width/height/top/left. All are arguably "hidden" in the parlance of our API documentation, so our handling comes down to a matter of taste (which is responsible for the churn here; sorry about that @NekR):
* assume any data is valid and always apply scroll offset
  * inaccurate for disconnected/display:none
* assume zero width and zero height indicate disconnected/display:none; return `undefined` or `null` or `{ top: 0, left: 0 }`
  * inaccurate for rare zero-size edge cases
* consider zero width _or_ zero height "hidden"; return `undefined` or `null` or `{ top: 0, left: 0 }`
  * inaccurate for slightly less rare zero-size edge cases

All the "inaccurate" results are allowed by our documentation, but are probably not equally useful to consumers (and in fact, the `undefined` or `null` results may be beneficial). Option 2 is only relevant if we want to give normal treatment to _single_-dimension elements (e.g., explicitly-sized elements or no-content display:block), so what's more valuable: accurate position for zero-size elements (first option), or distinct output for probably-unpositionable input (third or second option, especially with non-rect return)?

@dmethvin, @mzgol, @timmywil: care to weigh in?